mlrun/datastore/base.py

```diff
@@ -515,6 +515,15 @@ class HttpStore(DataStore):
     def __init__(self, parent, schema, name, endpoint="", secrets: dict = None):
         super().__init__(parent, name, schema, endpoint, secrets)
         self.auth = None
+        self._headers = None
+        if self._get_secret_or_env("HTTPS_AUTH_TOKEN"):
+            if schema == "https":
+                token = self._get_secret_or_env("HTTPS_AUTH_TOKEN")
+                self._headers = {"Authorization": f"token {token}"}
+            else:
+                raise mlrun.errors.MLRunInvalidArgumentError(
+                    "For using HTTPS_AUTH_TOKEN please use https schema"
+                )
```

Comment:
```
token = self._get_secret_or_env("HTTPS_AUTH_TOKEN")
if token:
  if schema not in ["https"]:
    raise blabla()
  else:
    self._headers = {"blabla": token}
```
we do the "get" once to ensure that if "get" really go externally and query for the secret, then do it once.

secondly, I'd move this entirely to 2 separate methods. "enrich" and "validate"

```
def __init__(...):
  ...
  self._enrich()
  self._validate()

def _enrich():
  token = self._get_secret_or_env("HTTPS_AUTH_TOKEN")
  if token:
    self._headers = {} if self._headers is None else self._headers
    self._headers.setdefault("Authorization", f"token {token}")

def _validate():
  if "Authorization" in self._headers and self._schema not in ["https"]:
    raise blabla()
```
  
more coherent and provide great room for extenions.